classes/Visualizer/Render/Library.php

```diff
@@ -206,7 +206,52 @@ private function getDisplayForm() {
 		</form>
 		</div>';
 	}
+	/**
+	 * Renders pro charts blocker.
+	 *
+	 * @access private
+	 */
+	private function _renderProPopupBlocker() {
+		$license = get_option( 'visualizer_pro_license_data', 'free' );
+		$license_key = '';
+		$download_id = '';
+		if ( ! empty( $license ) && is_object( $license ) ) {
+			$license_key = $license->key;
+			$download_id = $license->download_id;
+		}
+		$admin_license_url = admin_url( 'options-general.php#visualizer_pro_license' );
+		$renew_license_url = tsdk_utmify( Visualizer_Plugin::STORE_URL . '?edd_license_key=' . $license_key . '&download_id=' . $download_id, 'visualizer_license_block' );
 
+		if ( ! Visualizer_Module::is_pro() ) {
+			echo '
+				<div class="vizualizer-renew-notice-overlay" id="overlay-visualizer"></div>
+				<div class="vizualizer-renew-notice-popup">
+					<h1 class="vizualizer-renew-notice-heading">Alert!</h1>
+					<p class="vizualizer-renew-notice-message">' . esc_html__( 'In order to edit premium charts, benefit from updates and support for Visualizer Premium plugin, please renew your license code or activate it.', 'visualizer' ) . '	</p>
+					<div class="vizualizer-renew-notice-buttons-container">
+						<a href="' . esc_url( $renew_license_url) . '" target="_blank">
+							<button class="vizualizer-renew-notice-button vizualizer-renew-notice-renew-button">
+								<span class="dashicons dashicons-cart"></span>' . esc_html__( 'Renew License', 'visualizer' ) . ' 
+							</button>
+						</a>
+						<a href="' . esc_url( $admin_license_url ) . '">
+							<button class="vizualizer-renew-notice-button vizualizer-renew-notice-activate-button">
+								<span class="dashicons dashicons-unlock"></span> ' . esc_html__( 'Activate License', 'visualizer' ) . ' 
+							</button>
+						</a>
+					</div>
+					<i class="dashicons dashicons-no vizualizer-renew-notice-close-icon"></i>
+				</div>
+				<script>
+				jQuery(document).ready(function($) {
+					$(".vizualizer-renew-notice-close-icon").on("click", function() {
+						$("#overlay-visualizer").css("display", "none");
+						$(".vizualizer-renew-notice-popup").css("display", "none");
+					});
+				});
+				</script>';
```

Comment:
`.vizualizer-renew-notice-close-icon` can be wrapped in a `<button>` tag so that it is accessible and directly links a function on click that you define inside the `<script>` to avoid having a listener. You should also add some CSS to clear the button styles that are not used.